### PR TITLE
DL-184 celery director task level queues and options

### PR DESCRIPTION
**Pull request by CodePint**
Adds functionality to allow task options and kwargs to be set within a workflow definition on a task by task basis i.e:
default params, worker queues , task retries and backoff etc

- Updated WorkflowBuilder to enable complex workflows alongside existing simple workflows in yaml and json
- Allowing defaults to be set on the top level of a workflow/group which can be overridden by task level settings
- adding an additional test in test_builder.py for complex workflow with groups, retries, task level queues
- adding additional tests for complex workflows in test_workflows.py
- adding complex workflow to tests/workflows.yaml
- small helper/qol change in sqlalchemy base model (save returns self)
- adding queue to Task sqlalchemy model and generating the corresponding migration
- displaying Task queue from the frontend in index.html

ref: https://linius.atlassian.net/browse/DL-184

### CODE REVIEW

File: director/builder.py [hunk 1]
```diff
@@ -11,82 +13,85 @@
 
 class WorkflowBuilder(object):
     def __init__(self, workflow_id):
+        self.workflow = Workflow.query.filter_by(id=workflow_id).first()
```
Comment (CodePint) [inline director/builder.py]: switch to cached_property

File: director/builder.py [hunk 2]
```diff
@@ -11,82 +13,85 @@
...
+        self.blueprint = cel_workflows.get_by_name(self.workflow)
```
Comment (AlanDonohoe) [inline director/builder.py]: Still think "schema" might be a better name, and make more sense to developers... maybe...

File: director/builder.py [hunk 3]
```diff
@@ -11,82 +13,85 @@
...
-                    raise WorkflowSyntaxError()
-
-                sub_canvas_tasks = [
-                    self.new_task(t, single=False) for t in task[name]["tasks"]
-                ]
-
-                sub_canvas = group(*sub_canvas_tasks, task_id=uuid())
+                continue
+            if item["type"] == "group":
+                sub_canvas = self.new_group(item)
```
Comment (AlanDonohoe) [inline director/builder.py:78]: If all this is dealing with old (simple) vs new (complex) blueprints/schemas, and adding quite a bit of complexity in the code... why don't we just recreate the old blueprints/schemas in the new complex format?

Then we can remove all the if complex? And not have to refer to each entry as a v generic named "item", but instead a task_group? Or similar...?

File: director/builder.py [hunk 4]
```diff
@@ -11,82 +13,85 @@
...
                 canvas.append(sub_canvas)
-                self.previous = [s.id for s in sub_canvas_tasks]
-            else:
-                raise WorkflowSyntaxError()
-
+                continue
```
Comment (AlanDonohoe) [inline director/builder.py:86]: If all this is dealing with old (simple) vs new (complex) blueprints/schemas, and adding quite a bit of complexity in the code... why don't we just recreate the old blueprints/schemas in the new complex format?

Then we can remove all the if complex? stuff... And not have to refer to each entry as a v generic named "item", but instead a task_group? Or similar...?

**Pull request merged**